### ISSUE TITLE
Crash everytime on Xiaomi Mi4 , Google Pixel 2

### ISSUE THREAD

**Issue by hungvu193** [felipecsl/ktnes#25]
Can't not load nes file, just black screen, and when if i press any button it will crash.

```
06-05 12:58:30.643 6634-7180/com.felipecsl.knes A/art: art/runtime/runtime.cc:366]   at java.lang.Integer.valueOf(Integer.java:742)
06-05 12:58:30.643 6634-7180/com.felipecsl.knes A/art: art/runtime/runtime.cc:366]   at com.felipecsl.knes.INESFileParser$Companion.parseCartridge(INESFileParser.kt:49)
06-05 12:58:30.643 6634-7180/com.felipecsl.knes A/art: art/runtime/runtime.cc:366]   at com.felipecsl.knes.Director.<init>(Director.kt:13)
06-05 12:58:30.643 6634-7180/com.felipecsl.knes A/art: art/runtime/runtime.cc:366]   at com.felipecsl.knes.Director.<init>(Director.kt:10)
06-05 12:58:30.643 6634-7180/com.felipecsl.knes A/art: art/runtime/runtime.cc:366]   at com.felipecsl.knes.app.MainActivity.startConsole(MainActivity.kt:122)
06-05 12:58:30.643 6634-7180/com.felipecsl.knes A/art: art/runtime/runtime.cc:366]   at com.felipecsl.knes.app.MainActivity.onClickPlayPause(MainActivity.kt:89)
06-05 12:58:30.643 6634-7180/com.felipecsl.knes A/art: art/runtime/runtime.cc:366]   at com.felipecsl.knes.app.MainActivity.access$onClickPlayPause(MainActivity.kt:21)
06-05 12:58:30.643 6634-7180/com.felipecsl.knes A/art: art/runtime/runtime.cc:366]   at com.felipecsl.knes.app.MainActivity$onCreate$1.onClick(MainActivity.kt:67)
06-05 12:58:30.643 6634-7180/com.felipecsl.knes A/art: art/runtime/runtime.cc:366]   at android.view.View.performClick(View.java:5215)
06-05 12:58:30.643 6634-7180/com.felipecsl.knes A/art: art/runtime/runtime.cc:366]   at android.view.View$PerformClick.run(View.java:21193)
06-05 12:58:30.643 6634-7180/com.felipecsl.knes A/art: art/runtime/runtime.cc:366]   at android.os.Handler.handleCallback(Handler.java:742)
06-05 12:58:30.643 6634-7180/com.felipecsl.knes A/art: art/runtime/runtime.cc:366]   at android.os.Handler.dispatchMessage(Handler.java:95)
```

**Comment by felipecsl** [comment]
Do you have a NES rom file in the `res/raw` directory?

**Comment by hungvu193** [comment]
yes, i have

**Comment by felipecsl** [comment]
do you mind sharing here the rom file please?

**Comment by hungvu193** [comment]
yeah, i downloaded it from here 
https://romsmania.cc/download/roms/nintendo/super-mario-bros-242404

**Comment by brandoFS** [comment]
I'm getting a crash on all my devices. Emulator and Pixel/Pixel 2 XL

Same thing, I have a ROM in the raw folder, I get a black screen and crash when anything happens

**Comment by brandoFS** [comment]
2019-06-16 15:53:24.234 11381-11381/com.felipecsl.knes E/InputEventReceiver: Exception dispatching input event.
2019-06-16 15:53:24.234 11381-11381/com.felipecsl.knes E/MessageQueue-JNI: Exception in MessageQueue callback: handleReceiveCallback
2019-06-16 15:53:24.236 11381-11381/com.felipecsl.knes E/MessageQueue-JNI: kotlin.UninitializedPropertyAccessException: lateinit property director has not been initialized
        at com.felipecsl.knes.app.MainActivity.access$getDirector$p(MainActivity.kt:21)
        at com.felipecsl.knes.app.MainActivity$onButtonTouched$1$1.onTouch(MainActivity.kt:45)
        at android.view.View.dispatchTouchEvent(View.java:11772)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:448)
        at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1829)
        at android.app.Activity.dispatchTouchEvent(Activity.java:3307)
        at androidx.appcompat.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:69)
        at androidx.appcompat.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:69)
        at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:410)
        at android.view.View.dispatchPointerEvent(View.java:12015)
        at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:4795)
        at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4609)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4147)
        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4200)
        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4166)
        at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4293)
        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4174)
        at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4350)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4147)
        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4200)
        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4166)
        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4174)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4147)
        at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:6661)
        at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:6635)
        at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:6596)
        at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:6764)
        at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:186)
        at android.os.MessageQueue.nativePollOnce(Native Method)
        at android.os.MessageQueue.next(MessageQueue.java:325)
        at android.os.Looper.loop(Looper.java:142)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
    	at com.android.intern
2019-06-16 15:53:24.237 11381-11381/com.felipecsl.knes D/AndroidRuntime: Shutting down VM
    
    
    --------- beginning of crash
2019-06-16 15:53:24.239 11381-11381/com.felipecsl.knes E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.felipecsl.knes, PID: 11381
    kotlin.UninitializedPropertyAccessException: lateinit property director has not been initialized
        at com.felipecsl.knes.app.MainActivity.access$getDirector$p(MainActivity.kt:21)
        at com.felipecsl.knes.app.MainActivity$onButtonTouched$1$1.onTouch(MainActivity.kt:45)
        at android.view.View.dispatchTouchEvent(View.java:11772)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2968)
        at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2600)
        at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:448)
        at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1829)
        at android.app.Activity.dispatchTouchEvent(Activity.java:3307)
        at androidx.appcompat.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:69)
        at androidx.appcompat.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:69)
        at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:410)
        at android.view.View.dispatchPointerEvent(View.java:12015)
        at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:4795)
        at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4609)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4147)
        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4200)
        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4166)
        at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4293)
        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4174)
        at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4350)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4147)
        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4200)
        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4166)
        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4174)
        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4147)
        at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:6661)
        at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:6635)
        at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:6596)
        at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:6764)
        at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:186)
        at android.os.MessageQueue.nativePollOnce(Native Method)
        at android.os.MessageQueue.next(MessageQueue.java:325)
        at android.os.Looper.loop(Looper.java:142)
2019-06-16 15:53:24.240 11381-11381/com.felipecsl.knes E/AndroidRuntime:     at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
    
    
    --------- beginning of system

**Comment by felipecsl** [comment]
Fixed via 2182d5c496c8a5dd77ae0b896fa059b6d72636da

You're supposed to press the "play" button at the bottom right corner of the screen first to start the emulator, then other controls will be active.

Please let me know if you still can't play after following this step
